core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -79,11 +80,15 @@ private class ClientActor(driverArgs: ClientArguments, conf: SparkConf)
           driverArgs.supervise,
           command)
 
+      for (masterActor <- mastersActor) {
         masterActor ! RequestSubmitDriver(driverDescription)
+      }
```

Comment:
I don't think we want to submit to all masters right? I believe HA means we want to failover to backup Masters in case the primary one fails.